src/lib/pages/validator-details/components/validator-overview/VotingPowerOverview.tsx

```diff
@@ -1,46 +1,126 @@
-import { Flex, Heading, Text } from "@chakra-ui/react";
+import { Flex, Grid, Heading, Text } from "@chakra-ui/react";
+import type { BigSource } from "big.js";
+import Big from "big.js";
 
+import { TokenImageRender } from "lib/components/token";
 import { ValueWithIcon } from "lib/components/ValueWithIcon";
-
-// TODO remove mockup data
+import { getUndefinedTokenIcon } from "lib/pages/pools/utils";
+import type { AssetInfo, AssetInfos, Option, Token, U, USD } from "lib/types";
+import {
+  calculateAssetValue,
+  divWithDefault,
+  formatPrettyPercent,
+  formatPrice,
+  formatUTokenWithPrecision,
+  getTokenLabel,
+  toToken,
+} from "lib/utils";
 
 const VotingPowerDetail = ({
   label,
   percent,
   amount,
   denom,
-  value,
+  assetInfo,
 }: {
   label: string;
-  percent: string;
-  amount: string;
-  denom: string;
-  value: string;
-}) => (
-  <Flex direction="column" w="full">
-    <Text variant="body2" color="text.dark" mb={2}>
-      {label}
-    </Text>
-    <Text fontWeight={700} variant="body1">
-      {percent}
-    </Text>
-    <Text fontWeight={700} variant="body2">
-      {amount}{" "}
-      <span
-        style={{
-          fontWeight: "400",
-        }}
-      >
-        {denom}
-      </span>
-    </Text>
-    <Text variant="body3" color="text.dark">
-      ({value})
-    </Text>
-  </Flex>
-);
+  percent: number;
+  amount: U<Token<Big>>;
+  denom: Option<string>;
+  assetInfo: Option<AssetInfo>;
+}) => {
+  const formattedPercent = formatPrettyPercent(percent, 2, true);
+  const formattedAmount = formatUTokenWithPrecision(
+    amount,
+    assetInfo?.precision ?? 0,
+    true,
+    2
+  );
+  const formattedValue = assetInfo
+    ? formatPrice(
+        calculateAssetValue(
+          toToken(amount as U<Token<BigSource>>, assetInfo.precision),
+          assetInfo.price as USD<number>
+        )
+      )
+    : undefined;
+
+  return (
+    <Flex direction="column" w="full">
+      <Text variant="body2" color="text.dark" mb={2}>
+        {label}
+      </Text>
+      <Text fontWeight={700} variant="body1">
+        {formattedPercent}
+      </Text>
+      <Text fontWeight={700} variant="body2">
+        {formattedAmount}{" "}
+        {denom && (
+          <span
+            style={{
+              fontWeight: "400",
+            }}
+          >
+            {getTokenLabel(denom, assetInfo?.symbol)}
```

Comment:
```
add space here like below instead
` {getTokenLabel(denom, assetInfo?.symbol)}`
```

Reply:
It not possible to do ` {getTokenLabel(denom, assetInfo?.symbol)}` as spacebar na, since it wrap by HTML tag. But {` ${getTokenLabel(denom, assetInfo?.symbol)}`} is possible. Anyway, I will go with 
```js
{denom && (
          <span
            style={{
              fontWeight: "400",
            }}
          >
            {" "}
            {getTokenLabel(denom, assetInfo?.symbol)}
          </span>
        )}
```